map/bookmark_manager.cpp

```diff
@@ -404,6 +404,59 @@ void BookmarkManager::ResetRecentlyDeletedBookmark()
   m_recentlyDeletedBookmark.reset();
 }
 
+size_t BookmarkManager::GetRecentlyDeletedCategoriesCount() const
+{
+  Platform::FilesList files;
+  Platform::GetFilesByExt(GetTrashDirectory(), kKmlExtension, files);
+  return files.size();
+}
+
+BookmarkManager::KMLDataCollectionPtr BookmarkManager::GetRecentlyDeletedCategories()
```

Comment:
Is this method called synchronously from the main thread? It may cause a huge delay for large deleted lists in such a case. Please also make sure that it's only called when necessary (when Recently Deleted UI is displayed).

Reply:
> Is this method called synchronously from the main thread?

Yes

>  It may cause a huge delay for large deleted lists in such a case. Please also make sure that it's only called when necessary (when Recently Deleted UI is displayed).

This is why the `GetRecentlyDeletedCategoriesCount` was added.

The `GetRecentlyDeletedCategories` was called only when the `RecentlyDeletedCategoriesViewController` is opening and during the it's updates (when the category was deleted by the cloud for example).